CHANGELOG.md

```diff
@@ -17,11 +17,11 @@ The changes are relative to the previous release, unless the baseline is specifi
   draft.
 * Ignore gain maps with unsupported metadata. Handle gain maps with
   writer_version > 0 correctly.
-  The combination of settings enableParsingGainMapMetadata=false with
-  enableDecodingGainMap=true is no longer allowed and returns an invalid argument
-  error.
-  The `gainMapPresent` field is now only populated if enableParsingGainMapMetadata
-  is true.
+  Simplify gain map API: remove the enableParsingGainMapMetadata setting, now gain
+  map metadata is always parsed if present and if this feature is compiled in.
```

Comment:
Since gain map metadata is always parsed if present, we should write gain map metadata early in the file.

In src/write.c, we have

```
static avifResult avifEncoderWriteMediaDataBox(avifEncoder * encoder,
                                               avifRWStream * s,
                                               avifEncoderItemReferenceArray * layeredColorItems,
                                               avifEncoderItemReferenceArray * layeredAlphaItems)
{
    ...
    for (uint32_t itemPasses = 0; itemPasses < 3; ++itemPasses) {
        // Use multiple passes to pack in the following order:
        //   * Pass 0: metadata (Exif/XMP)
        //   * Pass 1: alpha, gain map (AV1)
        //   * Pass 2: all other item data (AV1 color)
```

Should we move gain map from Pass 1 to Pass 0 or even a new pass before the current Pass 0?

Reply:
This was done in https://github.com/AOMediaCodec/libavif/pull/2479